api/schedule.go

```diff
@@ -0,0 +1,42 @@
+package api
+
+import (
+	"time"
+)
+
+type Timetable struct {
+	PerHour    uint     `json:"per-hour"`
+	HoursOfDay []uint   `json:"hours-of-day"`
+	DaysOfWeek []string `json:"days-of-week"`
+}
+
+type Actor struct {
+	ID    string `json:"id"`
+	Login string `json:"login"`
+	Name  string `json:"name"`
+}
+
+type Schedule struct {
+	ID          string            `json:"id"`
+	ProjectSlug string            `json:"project_slug"`
+	Name        string            `json:"name"`
+	Description string            `json:"description,omitempty"`
+	Timetable   Timetable         `json:"timetable"`
+	Actor       Actor             `json:"actor"`
+	Parameters  map[string]string `json:"parameters"`
+	CreatedAt   time.Time         `json:"created_at"`
+	UpdatedAt   time.Time         `json:"updated_at"`
+}
+
+type ScheduleInterface interface {
+	Schedules(vcs, org, project string) (*[]Schedule, error)
```

Comment:
❓  should we add `ctx context.Context` ad the first argument to all of these functions? 

Will whatever is calling this ever care about us honouring timeouts or cancellations of requests that could be in-flight?

Reply:
In this case it seems that neither the CLI nor the terraform provider seem to, at least in existing code.

Reply:
Very fair, it's not worth adding in this case then